src/core/load-fragment.c

```diff
@@ -2860,7 +2861,15 @@ int config_parse_device_allow(
                 return 0;
         }
 
-        n = strcspn(rvalue, WHITESPACE);
+        r = unit_full_printf(userdata, rvalue, &t);
+        if(r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to resolve specifiers in %s, ignoring: %m",
+                           rvalue);
+        }
+
+        n = strcspn(t, WHITESPACE);
+
         path = strndup(rvalue, n);
```

Comment:
sorry is missed this earlier, but you need to write `t` here too, not `rvalue`...